### ISSUE TITLE
End User Portal 1.0.13  - Widget custom color lost on submission

### ISSUE THREAD

**Issue by foliovision** [freescout-helpdesk/freescout#1244]
We use a code like this to put up the widget on our website:

```
<script>
var FreeScoutW={
  s:{"settings":{"footer":"© {%year%} {%mailbox.name%}"},"color":"#bd0025","position":"br","id":...}
};
(function(d,e,s){
  if(d.getElementById("freescout-w"))return;
  a=d.createElement(e);
  m=d.getElementsByTagName(e)[0];
  a.async=1;
  a.id="freescout-w";
  a.src=s;
  m.parentNode.insertBefore(a, m)
})
(document,"script","https://site.com/modules/enduserportal/js/widget.js?v=3572");
</script>
```

The widget shows an an iframe with URL: `https://site.com/help/widget/form/2213982364?settings=%5Bobject%20Object%5D&color=%23bd0025&position=br&id=2213982364`

So it includes that custom color set to #bd0025.

However once you submit the message, it goes to just: `https://site.com/help/widget/form/2213982364?success=1`

So the color in the URL is lost and it shows with the default blue color.

Could you please fix this?

**Comment by foliovision** [comment]
These are showstopper bugs for us as blue will not work with our colour scheme at all so we are unable to use the Custom Contact Form (which is supposed to be our alternative for Live Chat when we are not there).

**Comment by freescout-helpdesk** [comment]
We'll check it.

**Comment by freescout-helpdesk** [comment]
Fixed. Make sure to update End-User Portal and Chat module to the latest version.

**Comment by foliovision** [comment]
Hi Freescout Team,

Thanks for fixing the colours issue. We were doing some additional testing of live chat in production and ran into two other serious issues:

* too slow (delay between posting)
* no notifications either in OS, browser or behind bell (the notifications for live chat should be separate category on the notifications setting screen for each user: https://desk.domain.com/users/notifications/1

Active live chats should not be like normal email threads but happen in real time in a chat like environment. Finished chats can be turned into normal tickets when closed. I would suggest that a live chat should just

Additional live chats with the same person could then become another ticket. I know that would break the simple structure of answer/reply by a single individual but it would make for useful live chat. Saving each chat comment as a separate answer would create too much chrome within a thread.

The Client should remain the initiator of the chat but there could certainly be a LiveChat persona, preferably for each separate live chat instance set up (I believe one can have a separate chat agent for each mailbox). I'd be happy to interact with you in more detail about improving the live chat and even drum up some screenshots if it would help you.

Right now we've retreated to using RocketChat as live chat when we are live and switching to FreeScout contact form in offer hours. It's easy to switch as we've built a custom single button at the top of the admin section of our website to take chat live or put the contact form up.

We were having trouble with the RocketChat contact form (RocketChat would no longer switch to contact form and would just leave chatters hanging) so this setup is the best of both world for now. Finished chats get mailed into our FreeScout support mailbox in case further follow up is needed. We'd be happy to go all FreeScout live chat, if and when the live chat is improved.

Great work adding so many powerful features and modules. We love the formatting module for instance, makes life much easier.
